clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -195,6 +195,25 @@ public static File tempDirectory() {
         return tempDirectory(null);
     }
 
+    /**
+     * Create a temporary directory under the given root directory.
+     * The root directory is removed on JVM exit if it doesn't already exist
+     * when this function is invoked.
+     *
+     * @param root path to create temporary directory under
+     * @return
+     */
+    public static File tempRelativeDir(String root) {
+        File rootFile = new File(root);
+        boolean created = rootFile.mkdir();
+
+        File result = tempDirectory(rootFile.toPath(), null);
+        if (created) {
+            rootFile.deleteOnExit();
```

Comment:
Force pushed the updates, PTAL again

Reply:
This is not enough to ensure the file is removed - it will only happen if the directory is empty.

Reply:
> This is not enough to ensure the file is removed - it will only happen if the directory is empty.

the sub-folder created by `tempDirectory(rootFile.toPath(), null)` will get removed so the `rootFile` should be empty and it is removable when exit. If callers want to create another file under passed `root` manually, they should not use this helper.

Reply:
> Have you verified that the order of these independent removals is correct though? 

I had run the tests with this PR, and the folder created by this helper gets removed as expected. see https://github.com/apache/kafka/pull/15289#pullrequestreview-1923156750